### PR TITLE
feat: database layer for local profile storage

### LINKED ISSUE #279: Add data layer for local storage

### PR DESCRIPTION
**Pull request by rami-mallah**
> Disclaimer: This PR adds 800 lines, but only a little less than 200 lines are new code. The rest of the lines are documentation and tests. And even this code follows the exact same pattern as `RentalListingEntity`, `ReviewEntity` and `RentalListingDao`, `ReviewDao`. 

## Summary
This PR adds Room database infrastructure to store the current user's profile locally for offline access. This PR focuses solely on the database layer (entities, DAO, migrations, and tests). The repository layer implementation will follow in a next PR. This closes #279.

## Implementation
- ProfileEntity.kt: Room entity that flattens `Profile`, `UserInfo`, and `UserSettings` into a single table with conversion methods
- ProfileDao.kt: Data Access Object with CRUD operations for the current user's profile
- AppDatabase.kt: Added `ProfileEntity`, incremented version to 3, added migration from 2 to 3

### Key Design:
- Only the current logged-in user's profile is stored locally
- Nested structures flattened into single table; enums stored as comma-separated strings
- Database migration handles existing users automatically

## Tests

Added Unit tests:
- ProfileEntityTest.kt: tests covering entity conversion, edge cases, enum handling, and null safety
- ProfileDaoTest.kt: tests covering CRUD operations, type preservation, and null handling

## Next steps
The following PRs will implement:
- `ProfileRepositoryLocal` - Room-backed repository implementation
- `ProfileRepositoryHybrid` - Hybrid repository with offline/online fallback
- Profile sync logic in ViewModels (on login, profile load/edit)
- Integration with `ProfileRepositoryProvider`

## Notes
- This PR does not change any existing functionality.
- All documentation and tests were generated with the help of AI.
- The SQL code for the migration was generated using AI.

### CODE REVIEW

File: app/src/main/java/com/android/mySwissDorm/model/database/ProfileEntity.kt [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            maxSize = maxSize,
+            preferredRoomTypes = roomTypes,
+            bookmarkedListingIds = bookmarkedListingIds)
+
+    val userSettings =
+        UserSettings(
+            language = languageEnum,
+            isPublic = isPublic,
+            isPushNotified = isPushNotified,
+            darkMode = darkMode)
```
Comment (ghassanbaroudi) [inline app/src/main/java/com/android/mySwissDorm/model/database/ProfileEntity.kt]: good separation between elements that are unique to the profile and settings

File: app/src/main/java/com/android/mySwissDorm/model/database/ProfileEntity.kt [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+          maxPrice = profile.userInfo.maxPrice,
+          minSize = profile.userInfo.minSize,
+          maxSize = profile.userInfo.maxSize,
+          preferredRoomTypes = roomTypesString,
+          bookmarkedListingIds = profile.userInfo.bookmarkedListingIds,
+          language = profile.userSettings.language.name,
+          isPublic = profile.userSettings.isPublic,
+          isPushNotified = profile.userSettings.isPushNotified,
+          darkMode = profile.userSettings.darkMode)
+    }
```
Comment (ghassanbaroudi) [inline app/src/main/java/com/android/mySwissDorm/model/database/ProfileEntity.kt]: LGTM
